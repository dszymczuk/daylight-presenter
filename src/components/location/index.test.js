import React from "react";
import { render, cleanup } from '@testing-library/react';

import Location from "./";
import LocationContextProvider from "../../contexts/locations";

afterEach(cleanup);

it("renders Location", () => {
  const props = {
    name: "Sample location",
    lat: 20,
    lng: 30,
    date: '01.01.2020',
    index: 0,
  };

  const { asFragment } = render(
    <LocationContextProvider>
      <Location {...props} />
    </LocationContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});