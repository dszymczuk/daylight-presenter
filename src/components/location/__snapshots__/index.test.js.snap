// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Location 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <h5
      class="card-header"
    >
      <div
        class="header"
      >
        <div>
          Sample location 
          <span>
            (20, 30)
          </span>
        </div>
        <div>
          01.01.2020
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            Edit
          </button>
          <button
            class="btn btn-danger btn-sm"
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
    </h5>
    <div
      class="card-body"
    >
      <div
        class="progressBarWrapper"
      >
        <div
          class="progress"
        >
          <div
            aria-valuemax="1440"
            aria-valuemin="0"
            aria-valuenow="NaN"
            class="progress-bar bg-warning"
            role="progressbar"
          />
          <div
            aria-valuemax="1440"
            aria-valuemin="0"
            aria-valuenow="NaN"
            class="progress-bar bg-success"
            role="progressbar"
          />
          <div
            aria-valuemax="1440"
            aria-valuemin="0"
            aria-valuenow="NaN"
            class="progress-bar bg-warning"
            role="progressbar"
          />
        </div>
        <div
          class="labels"
        >
          <div
            class="startHour"
          >
            00:00
          </div>
          <div
            class="endHour"
          >
            23:59
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
